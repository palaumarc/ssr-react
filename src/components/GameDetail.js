import React from "react";
import PropTypes from 'prop-types';

const GameDetail = ({ game, lastRun }) => (
    <div style={{width: 'fit-content', padding: '10px', border: '1px solid grey', display: 'grid', gridTemplateColumns: 'min-content max-content', placeItems: 'center', gridColumnGap: '5px'}} >
        <div style={{display: 'grid', gridRowGap: '10px', gridTemplateRows: '80% 20%', placeItems: 'center'}} >
            <img alt={game.name} src={game.logoUrl} />
            {game.name}
        </div>
        <div style={{display: 'grid', gridRowGap: '10px', gridTemplateRows: 'auto auto auto', placeItems: 'start / center'}} >
            <button style={{ cursor: 'pointer', border: 'none', textAlign: 'center', borderRadius: '4px', backgroundColor: '#008CBA', fontSize: '20px'}}><a style={{ color: 'white', textDecoration: 'none' }}href={lastRun.videoUrl}>Video</a></button>
            <span>Player: {lastRun.playerName}</span>
            <span>Time: {lastRun.duration.minutes()} m {lastRun.duration.seconds()} s</span>
        </div>
    </div>
)

GameDetail.propTypes = {
    game: PropTypes.shape({
        name: PropTypes.string,
        logoUrl: PropTypes.string,
      }).isRequired,
    lastRun: PropTypes.shape({
        playerName: PropTypes.string,
        videoUrl: PropTypes.string,
        duration: PropTypes.object
    }).isRequired,
  };

export default GameDetail;