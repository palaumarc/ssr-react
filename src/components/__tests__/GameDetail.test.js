import React from 'react';
import GameDetail from '../GameDetail';
import renderer from 'react-test-renderer';
import moment from 'moment';

it('All props are defined. Snapshot', () => {
    const game = {
        name: 'dummyName',
        logoUrl: 'dummyLogoUrl'
    } 

    const gameRun = {
        playerName: 'dummyPlayerName',
        videoUrl: 'dummyVideoUrl',
        duration: moment.duration(3, 'minute')
    }

    const tree = renderer.create(<GameDetail game={game} lastRun={gameRun} />).toJSON();
    expect(tree).toMatchSnapshot();
});