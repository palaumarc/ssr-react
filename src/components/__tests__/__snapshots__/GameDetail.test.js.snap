// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props are defined. Snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid grey",
      "display": "grid",
      "gridColumnGap": "5px",
      "gridTemplateColumns": "min-content max-content",
      "padding": "10px",
      "placeItems": "center",
      "width": "fit-content",
    }
  }
>
  <div
    style={
      Object {
        "display": "grid",
        "gridRowGap": "10px",
        "gridTemplateRows": "80% 20%",
        "placeItems": "center",
      }
    }
  >
    <img
      alt="dummyName"
      src="dummyLogoUrl"
    />
    dummyName
  </div>
  <div
    style={
      Object {
        "display": "grid",
        "gridRowGap": "10px",
        "gridTemplateRows": "auto auto auto",
        "placeItems": "start / center",
      }
    }
  >
    <button
      style={
        Object {
          "backgroundColor": "#008CBA",
          "border": "none",
          "borderRadius": "4px",
          "cursor": "pointer",
          "fontSize": "20px",
          "textAlign": "center",
        }
      }
    >
      <a
        href="dummyVideoUrl"
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        Video
      </a>
    </button>
    <span>
      Player: 
      dummyPlayerName
    </span>
    <span>
      Time: 
      3
       m 
      0
       s
    </span>
  </div>
</div>
`;
